hkmc2/shared/src/main/scala/hkmc2/codegen/HandlerLowering.scala

```diff
@@ -454,44 +599,61 @@ class HandlerLowering(paths: HandlerPaths, opt: EffectHandlers)(using TL, Raise,
-      companion = cls.companion.map(translateBody(_, cls.sym)))
+      ctor = translateBlock(cls.ctor, Set.empty, N, R(cls.isym), curCtorCtx),
+      companion = cls.companion.map(translateBody(_, cls.sym))) // TODO: callSelf
   
   // Handle block becomes a FunDefn and CallPlaceholder
   private def translateHandleBlock(h: HandleBlock)(using HandlerCtx): Block =
     val sym = BlockMemberSymbol(s"handleBlock$$", Nil)
     val lbl = freshTmp("handlerBody")
     val lblLoop = freshTmp("handlerLoop")
     
-    val handlerBody = translateBlock(h.body, Set.empty, HandlerCtx(false, true,
-      s"Cont$$handleBlock$$${symToStr(h.lhs)}$$", N, handlerCtx.debugInfo.copy(debugNme = s"‹handler body of ${h.lhs.nme}›"), state => blockBuilder
-        .assignFieldN(state.res.asPath.contTrace.last, nextIdent,
-          Instantiate(mut = true, state.cls, Value.Lit(Tree.IntLit(state.uid)).asArg :: Nil))
-        .ret(PureCall(paths.handleBlockImplPath, state.res.asPath :: h.lhs.asPath :: Nil))))
+    val handlerBody = translateBlock(
+      h.body, Set.empty, N, L(sym), // TODO: callSelf 
+      HandlerCtx(
+        false, true,
+        s"Cont$$handleBlock$$${symToStr(h.lhs)}$$", N, 
+        handlerCtx.debugInfo.copy(debugNme = s"‹handler body of ${h.lhs.nme}›"), 
+        state => blockBuilder
+          .assignFieldN(state.res.asPath.contTrace.last, nextIdent, Instantiate(true, state.cls, state.uid.asArg :: Nil))
+          .ret(PureCall(paths.handleBlockImplPath, state.res.asPath :: h.lhs.asPath :: Nil))
+      )
+    )
     
     val handlerMtds = h.handlers.map: handler =>
-      val lamSym = BlockMemberSymbol("lambda", Nil, false)
-      val lamDefn = FunDefn(
-        N,
-        lamSym,
-        PlainParamList(Param(FldFlags.empty, handler.resumeSym, N, Modulefulness.none) :: Nil) :: Nil,
-        translateBlock(handler.body,
-          handler.params.flatMap(_.paramSyms).toSet,
-          handlerMtdCtx(s"Cont$$handler$$${symToStr(h.lhs)}$$${symToStr(handler.sym)}$$", handler.sym.nme)))
+      val sym = BlockMemberSymbol("hdlrFun", Nil, true)
+      val mtdBdy = translateBlock(handler.body,
+        handler.params.flatMap(_.paramSyms).toSet, N, L(sym), // TODO: callSelf
+        handlerMtdCtx(s"Cont$$handler$$${symToStr(h.lhs)}$$${symToStr(handler.sym)}$$", handler.sym.nme))
+      val fDef = FunDefn(
+        N, sym, PlainParamList(Param(FldFlags.empty, handler.resumeSym, N, Modulefulness.none) :: Nil) :: Nil,
+        mtdBdy 
+        )
       FunDefn(
         S(h.cls),
         handler.sym, handler.params,
         Define(
-          lamDefn,
-          Return(PureCall(paths.mkEffectPath, h.cls.asPath :: Value.Ref(lamSym) :: Nil), false)))
+          fDef,
+          Return(PureCall(paths.mkEffectPath, h.cls.asPath :: Value.Ref(sym) :: Nil), false)))
+    
+    // some limit handling of effects extending classes and having access to their fields
+    // does not support super() raising effects
```

Comment:
I can't understand the grammar of these comment lines. Are they even full sentences?